ESPWebThingAdapter.h

```diff
@@ -567,84 +545,7 @@ class WebThingAdapter {
     request->send(response);
   }
 
-  void handleThingActionsGet(AsyncWebServerRequest *request,
-                             ThingDevice *device) {
-    if (!verifyHost(request)) {
-      return;
-    }
-    AsyncResponseStream *response =
-        request->beginResponseStream("application/json");
-
-    DynamicJsonDocument doc(LARGE_JSON_DOCUMENT_SIZE);
-    JsonArray queue = doc.to<JsonArray>();
-    device->serializeActionQueue(queue);
-    serializeJson(queue, *response);
-    request->send(response);
-  }
-
-  void handleThingActionsPost(AsyncWebServerRequest *request,
-                              ThingDevice *device) {
-    if (!verifyHost(request)) {
-      return;
-    }
-
-    if (!b_has_body_data) {
-      request->send(422); // unprocessable entity (b/c no body)
-      return;
-    }
-
-    DynamicJsonDocument *newBuffer =
-        new DynamicJsonDocument(SMALL_JSON_DOCUMENT_SIZE);
-    auto error = deserializeJson(*newBuffer, (const char *)body_data);
-    if (error) { // unable to parse json
-      b_has_body_data = false;
-      memset(body_data, 0, sizeof(body_data));
-      request->send(500);
-      delete newBuffer;
-      return;
-    }
-
-    JsonObject newAction = newBuffer->as<JsonObject>();
-
-    if (newAction.size() != 1) {
-      b_has_body_data = false;
-      memset(body_data, 0, sizeof(body_data));
-      request->send(400);
-      delete newBuffer;
-      return;
-    }
-
-    ThingActionObject *obj = device->requestAction(newBuffer);
-
-    if (obj == nullptr) {
-      b_has_body_data = false;
-      memset(body_data, 0, sizeof(body_data));
-      request->send(500);
-      delete newBuffer;
-      return;
-    }
-
-#ifndef WITHOUT_WS
-    obj->setNotifyFunction(std::bind(&ThingDevice::sendActionStatus, device,
-                                     std::placeholders::_1));
-#endif
-
-    DynamicJsonDocument respBuffer(SMALL_JSON_DOCUMENT_SIZE);
-    JsonObject item = respBuffer.to<JsonObject>();
-    obj->serialize(item, device->id);
-    String jsonStr;
-    serializeJson(item, jsonStr);
-    AsyncWebServerResponse *response =
-        request->beginResponse(201, "application/json", jsonStr);
-    request->send(response);
-
-    b_has_body_data = false;
-    memset(body_data, 0, sizeof(body_data));
-
-    obj->start();
-  }
```

Comment:
As mentioned above, for the gateway we left the existing actions API in place so as not to remove features, and only modified the payload of the `invokeaction` operation (i.e. `POST /actions/{actionName}`).

Reply:
Actions under `action/actionName` are supported. That isn't an issue. You can take a look into the examples.
I only removed the part that you can call `/action`. I don't know, if that ever worked. When I tried it, it broke everything. Since it is also not important, I just removed it.

Reply:
> Since it is also not important

If you remove the `/actions` endpoint then there is no way of getting a list of all actions currently running on the device. It might not be important to you, but it's not necessary to remove this feature in order for the Thing Description to be W3C compliant. I even added a [`queryallactions`](https://github.com/w3c/wot-thing-description/pull/1232) operation to the W3C Thing Description specification specifically for this purpose.